### PR TITLE
Fix document references in modules

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
~When looking at the existing references in documentation I found we did
not support references by document name. This adds this functionality
too and adds it to module **win_chocolatey**.~

This PR fixes:
- doc references by using proper internal references.
- rendering issue for module-entries with description
- rename reference with dash to reference with underscore

This will conflict with #49845, but I will fix it when needed.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs win_chocolatey

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 1]
```diff
@@ -247,6 +247,10 @@ All fields in the ``DOCUMENTATION`` block are lower-case. All fields are require
         - ref: aci_guide
           description: Detailed information on how to manage your ACI infrastructure using Ansible.
 
+        # Reference by rST document name
+        - doc: /user_guide/become
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst]: @acozine Do we want to use this form of referencing to a file name?
This means we can't move files around. I thought we should use `:ref:`

Comment (dagwieers) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst]:
```
[dag@moria ansible.git]$ grep -r ':doc:' docs/docsite/ | grep ':doc:' | wc -l
183
```

Comment (dagwieers) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst]:
```
[dag@moria ansible.git]$ grep -r ':doc:' docs/docsite/rst/modules/ | grep ':doc:' | wc -l
3
```

Comment (dagwieers) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst]: In this case jumping directly to "Become and Windows" would be even better.

**Comment by dagwieers** [comment]
Ok, I got rid of the doc references in the modules. However we still have plenty of those in the docsite.

**Comment by acozine** [comment]
Regarding `:doc:` vs. `:ref:` links, we have updated the Ansible style guide: https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#adding-internal-links
We want to all internal links to use the `:ref:` syntax.

**Comment by dagwieers** [comment]
@acozine Can you review this PR ? This gets rid of those in the modules, and updates the **win_chocolatey** module.

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst (hunk 1 above)
Comment (dagwieers) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst]: Fixed.

**Comment by acozine** [comment]
I didn't realize we could use `:ref:` directly in the module documentation. In https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-within-module-documentation we recommend using `L(link_text,../path/to/doc)`, which is the functional equivalent of a `:doc:` link.

**Comment by acozine** [comment]
Ah, no, the `L(link_text,URL)` construction doesn't seem to be intended for internal docs links at all. https://github.com/ansible/ansible/blob/devel/docs/bin/plugin_formatter.py#L122 and https://github.com/ansible/ansible/blob/devel/docs/bin/plugin_formatter.py#L143 simply offer a way to make the link text different from the URL itself. Interesting!

**Comment by dagwieers** [comment]
Correct, the `:ref:` references would only properly work for the website, but not for ansible-doc or anything else. But it has been done in the past.

We could introduce a `R(ref)` type of link, that would translate to an rST reference, or an HTML link that ends up in a search-engine, like we attempt to do for `seealso:` (I hope we can improve this mechanism at some point in the future).

**Pull request merged**